sycl/source/detail/error_handling/error_handling.cpp

```diff
@@ -230,6 +233,47 @@ void handleInvalidWorkGroupSize(const device_impl &DeviceImpl, pi_kernel Kernel,
           // else unknown.  fallback (below)
         }
       }
+    } else if (IsCuda) {
+      // CUDA:
+      // PI_ERROR_INVALID_WORK_GROUP_SIZE is returned when the kernel registers
+      // required for the launch config exceeds the maximum number of registers
+      // per block (PI_EXT_CODEPLAY_DEVICE_INFO_MAX_REGISTERS_PER_WORK_GROUP).
+      // This is if local_work_size[0] * ... * local_work_size[work_dim - 1]
+      // multiplied by PI_KERNEL_GROUP_INFO_NUM_REGS is greater than the value
+      // of PI_KERNEL_MAX_NUM_REGISTERS_PER_BLOCK. See Table 15: Technical
+      // Specifications per Compute Capability, for limitations.
+      const size_t TotalNumberOfWIs =
+          NDRDesc.LocalSize[0] * NDRDesc.LocalSize[1] * NDRDesc.LocalSize[2];
+
+      uint32_t NumRegisters = 0;
+      Plugin.call<PiApiKind::piKernelGetGroupInfo>(
+          Kernel, Device, PI_KERNEL_GROUP_INFO_NUM_REGS, sizeof(NumRegisters),
+          &NumRegisters, nullptr);
+
+      uint32_t MaxRegistersPerBlock = 0;
+      Plugin.call<PiApiKind::piDeviceGetInfo>(
+          Device, PI_EXT_CODEPLAY_DEVICE_INFO_MAX_REGISTERS_PER_WORK_GROUP,
+          sizeof(MaxRegistersPerBlock), &MaxRegistersPerBlock, nullptr);
```

Comment:
We might as well use the new info query here.
```suggestion
      uint32_t MaxRegistersPerBlock =
          DeviceImpl.get_info<ext::codeplay::experimental::info::device::
                                  max_registers_per_work_group>();
```